File: erlang/fib.erl
-module(fib).
-export([fibonacci/1]).

fibonacci(N) when N < 2 ->
    io:format("Computing fibonacci(~p)~n", [N]),
    N;
fibonacci(N) ->
    io:format("Computing fibonacci(~p)~n", [N]),
    fibonacci(N - 1) + fibonacci(N - 2).

File: elixir/fib.exs
defmodule Fib do
  def fibonacci(n) do
    IO.puts("Computing fibonacci(#{n})")
    cond do
      n < 2 -> n
      true -> fibonacci(n - 1) + fibonacci(n - 2)
    end
  end
end

IO.puts("Result: #{Fib.fibonacci(5)}")
